modules/deploy/pages/deployment-option/cloud/gcp-private-service-connect.adoc

```diff
@@ -0,0 +1,425 @@
+- `consumer_subnet_name`: The subnet where the clients will run. This cannot be a subnet with the *Purpose* set to `PRIVATE_SERVICE_CONNECT`. GCP doesn't allow such subnets to be reused for other purposes.
+
+[,bash]
+----
+cd terraform/2_consumer_resources
+cat << EOF > input.tfvars
+consumer_project_id = "<consumer-GCP-project-ID>"
+consumer_vpc_id = "<consumer-VPC-ID>"
+service_attachment_uri = "<service-attachment-uri>"
+consumer_region = "<consumer-region>"
+consumer_subnet_name = "<consumer-subnet-name>"
+EOF
+terraform init
+terraform apply -var-file=input.tfvars 
+----
+
+The module outputs a `psc_forwarding_rule_ip`. This is the IP through which the service attachment can be reached. To be able to use the Kafka API, as well as the HTTP Proxy subscriptions, you will need to set up the right private DNS zone and records. 
+
+=== Deploy the DNS zone and records
+
+Make sure to provide your own values for the following:
+
+- `consumer_project_id`: The ID of the GCP project where consumer (client) resources will be created. Should match one of the projects included in the `consumer_accept_list` field in previous steps.
+- `consumer_vpc_id`: The ID of the VPC where the consumers will be deployed.
+- `dns_a_records`: The list of DNS A records for each Redpanda broker API. These will be provided by the Redpanda team and will never change. Format: `["<record 1>", "<record 2>", "<record N>"]`. *All records must end with a dot*.
+- `dns_zone`: The base DNS zone for the records. Can be derived from them. *The DNS zone name must end with a dot*.
+- `dns_a_records_target_ip`: The IP address that the A records will point to, which should be that of the forwarding rule deployed in the previous step. You can query the target IP address with the following `gcloud` command:
++
+[,bash]
+----
+gcloud compute forwarding-rules describe redpanda-psc-endpoint --region <region> --project <consumer project ID>
+----
+
+[,bash]
+----
+cd terraform/3_dns
+cat << EOF > input.tfvars
+consumer_project_id = "<consumer GCP project ID>"
+consumer_vpc_id = "<consumer VPC ID>"
+dns_a_records = <list of broker A records>
+dns_zone = "<base Redpanda DNS zone>"
+dns_a_records_target_ip = "<PSC consumer forwarding rule IP>"
+EOF
+terraform init
+terraform apply -var-file=input.tfvars 
+----
+
+=== Update your firewall rules
+
+Make sure that your consumer VPC firewall allows traffic to and from the PSC forwarding rule IP address on the expected ports.
+
+== Test the connection
+
+You can test the PSC connection from any VM or container in the consumer VPC. If configuring a client isn't possible right away, you can do these checks using `rpk` or curl:
+
+=== Test Kafka API
+
+. Set the following environment variables.
++
+[,bash]
+----
+REDPANDA_BROKERS='<broker URL>:30292'
+REDPANDA_SASL_MECHANISM='<SCRAM-SHA-256 or SCRAM-SHA-512>'
+REDPANDA_SASL_USERNAME='<username>'
+REDPANDA_SASL_PASSWORD='<password>'
+----
+
+. Check the cluster status using `rpk`.
```

Comment:
Can we provide the curl equivalent of these commands?

Reply:
Yes. I'll test them and send them to you.

Reply:
Now that I think about it, the `curl` commands would just be the same in the [HTTP Proxy guide](https://docs.redpanda.com/current/develop/http-proxy/), changing the URL from `localhost:8082` to the PSC domain name shown in the UI.